docs/source/examples/mg-mode-part-ii.ipynb

```diff
@@ -0,0 +1,1684 @@
+    "\n",
+    "During this process, this notebook will illustrate the following features:\n",
+    "\n",
+    "   - Calculation of multi-group cross sections for a fuel assembly\n",
+    "   - Automated creation and storage of MGXS with openmc.mgxs.Library\n",
+    "   - Steady-state pin-by-pin fission rates comparison between continuous-energy and multi-group OpenMC.\n",
+    "   - Modification of the scattering data in the library to show the flexibility of the multi-group solver\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Generate Input Files"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "from IPython.display import Image\n",
+    "import matplotlib.pyplot as plt\n",
+    "import numpy as np\n",
+    "import os\n",
+    "\n",
+    "import openmc\n",
+    "\n",
+    "%matplotlib inline"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "First we need to define materials that will be used in the problem. Before defining a material, we must create nuclides that are used in the material."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# Instantiate some Nuclides\n",
+    "h1 = openmc.Nuclide('H1')\n",
+    "b10 = openmc.Nuclide('B10')\n",
+    "o16 = openmc.Nuclide('O16')\n",
+    "u235 = openmc.Nuclide('U235')\n",
+    "u238 = openmc.Nuclide('U238')\n",
+    "zr90 = openmc.Nuclide('Zr90')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "With the nuclides we defined, we will now create three materials for the fuel, water, and cladding of the fuel pins."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# 1.6% enriched fuel\n",
+    "fuel = openmc.Material(name='1.6% Fuel')\n",
+    "fuel.set_density('g/cm3', 10.31341)\n",
+    "fuel.add_nuclide(u235, 3.7503e-4)\n",
+    "fuel.add_nuclide(u238, 2.2625e-2)\n",
+    "fuel.add_nuclide(o16, 4.6007e-2)\n",
+    "\n",
+    "# zircaloy\n",
+    "zircaloy = openmc.Material(name='Zircaloy')\n",
+    "zircaloy.set_density('g/cm3', 6.55)\n",
+    "zircaloy.add_nuclide(zr90, 7.2758e-3)\n",
+    "\n",
+    "# borated water\n",
+    "water = openmc.Material(name='Borated Water')\n",
+    "water.set_density('g/cm3', 0.740582)\n",
+    "water.add_nuclide(h1, 4.9457e-2)\n",
+    "water.add_nuclide(o16, 2.4732e-2)\n",
+    "water.add_nuclide(b10, 8.0042e-6)\n"
```

Comment:
This material declaration can be simplified a lot.  You can pass a string directly to `add_nuclide` rather than first creating a Nuclide, and we can take advantage of the new `add_element(enrichment=...)` feature.
```python
fuel.add_element('U', 1.0, enrichment=1.6)
fuel.add_element('O', 2.0)
```
And with the `add_element` functionality, I see no reason to use only one nuclide of Zr
```python
zircaloy.add_element('Zr', 1)
```